cosmos/operators/virtualenv.py

```diff
@@ -97,6 +94,30 @@ def execute(self, context: Context) -> None:
             self._venv_tmp_dir.cleanup()
         logger.info(output)
 
+    def _get_or_create_venv_py_interpreter(self) -> str:
+        if self._venv_dir is not None:
+            py_interpreter_path = Path(f"{self._venv_dir}/bin/python")
+
+            self.log.info(f"Checking for venv interpreter: {py_interpreter_path} : {py_interpreter_path.is_file()}")
+            if py_interpreter_path.is_file():
+                self.log.info(f"Found Python interpreter in cached virtualenv: `{str(py_interpreter_path)}`")
+                return str(py_interpreter_path)
```

Comment:
Yeah ideally we'd be able to clean up the virtual env after the DAG run, but for the reasons you mentioned before this can be tricky. 
One way to "perhaps" invalidate the virtualenv is to check when it was created and, after say 24 hours or 48 or so, have this operator clean it up and recreate it?

Reply:
The time-based approach could lead to some strange scenarios and be tricky to troubleshoot. How feasible would be for us to run a `pip install` in an existing virtualenv? It should be very quick if it was previously setup, and it would make the operator reliable.

Regarding the cleanup - I know - ideally we'd be able to set the venv only once during the DAG setup and delete during tear down. Unfortunately - to my knowledge - even the latest Airflow (2.7) does not allow us to have a setup/tear down per worker node during the lifecycle of a DAG. But this can be an improvement for the future - in a separate PR!

Reply:
In our case, we probably would only need to do part of the logic:
```
    if requirements is not None and len(requirements) != 0:
        pip_cmd = _generate_pip_install_cmd_from_list(venv_directory, requirements, pip_install_options)
```
Since we don't support `requirements_file_path`.

If we don't want to add this call unnecessarily, we'd probably need a `pip freeze` call - to confirm if the desired dependencies are already installed, which may be more work.

We probably need one or both of these. Otherwise, we're at the risk of an Airflow worker having partial/outdated dependencies that are incompatible with the dependencies the user requested.

I'm in favour of us caching for performance reasons, but we still should aim to have the task being idempotent.